protobuf/coreapi/api.proto

```diff
@@ -42,6 +42,54 @@ service Core {
 
   // ServiceLogs gives a stream for dependency logs of a service.
   rpc ServiceLogs (ServiceLogsRequest) returns (stream LogData) {}
+
+  // CreateWorkflow creates and runs a new workflow.
+  rpc CreateWorkflow (CreateWorkflowRequest) returns (CreateWorkflowReply) {}
+
+  // DeleteWorkflow stops and deletes a workflow.
+  rpc DeleteWorkflow (DeleteWorkflowRequest) returns (DeleteWorkflowReply) {}
+}
+
+// The request's data for the `CreateWorkflow` API.
+//
+// **Example**
+// ```json
+// {
+//   "file":   __FILE__,
+//   "name": "__UNIQUE_WORKFLOW_NAME__"
+// }
+// ```
+message CreateWorkflowRequest {
+  bytes file = 1;  // workflow.yml file.
```

Comment:
This is sounds a better way. But, by extracting the yaml parsing logic from core, we're leaving this work to clients. _Workflow.yaml_ should be a spec defined by MESG and core should natively support that language. In this perspective, I like the idea of accepting yaml for workflow creations.

I don't think that we will ever need to compile a representation to another one like json to yaml, in fact, we may even need to create our own language in future and drop yaml for a much dedicated & cleaner syntax. 

For the `direct api call from a web ui`, I agree on the flexibility of this one but yaml will be widely used in forums discussions, tutorial pages or in mesg application market and community will get very familiar with it. Thus, application/yaml builders will be widely used and implemented by community as well. In this case, a JS library created by us or the community can build a yaml from a definition object that you're proposing.

@antho1404 I both like your version and the plain yaml support in the API. I'll be fine with both because both ideas have their own reasonings. As a reminder, on services, we also parse `mesg.yml` inside core instead leaving this parts to clients.

Reply:
For the `mesg.yml` we actually sent the whole directory to the core because there is a dependency between the files and the mesg.yml. We don't have this dependency here.

The reason why I want to be as close as possible as pure data and not the representation of these data is that we are not yet sure about the representation of the workflow file, we always talk about yml but we have to do more research, maybe there is existing workflow templates even maybe in xml that exists already with an existing ecosystem like an ui or some validation etc... If we keep only the data without the representation we can adapt really easily.

Also for the work from the client side, I don't see it as a problem, we have the cli in our case that can parse and send the data and an UI can directly send the data without having to create a fake temporary file just in order to send it to the api.

We could also have a specific language that is compiled to these data kind of like an AST for most programming languages and AST are the necessaries data not the representation of it. Representation are just for users (eg: prettify for js)

Reply:
Having API to accept an object instead yaml makes a bit more sense, I agree with it. In future, if we decide to create our own definition language, maybe then we can support both ways. So clients can build the object files or send the definition file directly depending on their cases. I'm updating the code!